src/components/app/details/cicdHistory/types.tsx

```diff
@@ -0,0 +1,208 @@
+import { TERMINAL_STATUS_MAP } from '../../../../config'
+import { OptionType } from '../../types'
+
+export interface WebHookData {
+    Id: number
+    EventActionType: string
+    Data: any
+}
+
+export interface History {
+    id: number
+    name: string
+    status: string
+    podStatus: string
+    message: string
+    startedOn: string
+    finishedOn: string
+    ciPipelineId: number
+    namespace: string
+    logLocation: string
+    gitTriggers: Map<number, GitTriggers>
+    ciMaterials: CiMaterial[]
+    triggeredBy: number
+    artifact: string
+    artifactId: number
+    triggeredByEmail: string
+    stage?: DeploymentStageType
+    blobStorageEnabled?: boolean
+}
+
+export interface CiMaterial {
+    id: number
+    gitMaterialId: number
+    gitMaterialUrl: string
+    gitMaterialName: string
+    type: string
+    value: string
+    active: boolean
+    lastFetchTime: string
+    isRepoError: boolean
+    repoErrorMsg: string
+    isBranchError: boolean
+    branchErrorMsg: string
+    url: string
+}
+
+export interface GitTriggers {
+    Commit: string
+    Author: string
+    Date: Date
+    Message: string
+    Changes: string[]
+    WebhookData: WebHookData
+    GitRepoUrl: string
+    GitRepoName: string
+    CiConfigureSourceType: string
+    CiConfigureSourceValue: string
+}
+
+export interface ArtifactType {
+    status: string
+    artifact: string
+    blobStorageEnabled: boolean
+    getArtifactPromise?: () => Promise<any>
+}
+
+export interface CopyTippyWithTextType {
+    copyText: string
+}
+
+export interface LogsRendererType {
+    triggerDetails: History
+    isBlobStorageConfigured: boolean
+    parentType: HistoryComponentType
+}
+
+export interface LogResizeButtonType {
+    fullScreenView: boolean
+    setFullScreenView: React.Dispatch<React.SetStateAction<boolean>>
+}
+
+export interface ScrollerType {
+    scrollToTop: (e: any) => void
+    scrollToBottom: (e: any) => void
+    style: any
```

Comment:
Can we please type this as CSSProperties?

Reply:
done